src/turndown.js

```diff
@@ -169,6 +169,29 @@ turndownService.addRule("inlinecodeblockfix", {
   }
 })
 
+/**
+ * turn <pre> elements into code blocks
+ */
+turndownService.addRule("code_block", {
+  filter:      node => node.nodeName === "PRE",
```

Comment:
We should combine this rule and the `codeblockfix` rule, leaving the filter as you have it here and matching on `pre` elements, then in the replacement rule having a special condition for `node.firstChild && node.firstChild.nodeName === "SPAN"`.  I'm fairly certain an element can only be matched to one turndown rule, and the case `codeblockfix` was meant to be solved is now being caught by this rule and not handled in the intended way.

Reply:
Yeah, this makes sense.
Thanks